docs/howtos/integrations/_langgraph_agent_evaluation.md

```diff
@@ -0,0 +1,550 @@
+     ToolMessage(content='0.0098', name='get_metal_price', id='0f29642b-b4e8-4b78-9923-007b87affbe2', tool_call_id='call_Ny714Y9Ql2NKGpBK4WJZI36H'),
+     AIMessage(content='The price of copper is $0.0098 per gram.', response_metadata={'token_usage': {'completion_tokens': 14, 'prompt_tokens': 148, 'total_tokens': 162, 'prompt_tokens_details': {'cached_tokens': 0, 'audio_tokens': 0}, 'completion_tokens_details': {'reasoning_tokens': 0, 'audio_tokens': 0, 'accepted_prediction_tokens': 0, 'rejected_prediction_tokens': 0}}, 'model_name': 'gpt-4o-mini-2024-07-18', 'system_fingerprint': 'fp_0ba0d124f1', 'finish_reason': 'stop', 'logprobs': None}, id='run-71526780-e9f9-4a79-a914-7863f8ec9078-0', usage_metadata={'input_tokens': 148, 'output_tokens': 14, 'total_tokens': 162})]
+
+
+
+### Converting Messages to Ragas Evaluation Format
+
+In the current implementation, the GraphState stores messages exchanged between the human user, the AI (LLM's responses), and any external tools (APIs or services the AI uses) in a list. Each message is an object in LangChain's format
+
+```python
+# Implementation of Graph State
+class GraphState(TypedDict):
+    messages: Annotated[list[AnyMessage], add_messages]
+```
+
+Each time a message is exchanged during agent execution, it gets added to the messages list in the GraphState. However, Ragas requires a specific message format for evaluating interactions.
+
+Ragas uses its own format to evaluate agent interactions. So, if you're using LangGraph, you will need to convert the LangChain message objects into Ragas message objects. This allows you to evaluate your AI agents with Ragas’ built-in evaluation tools.
+
+**Goal:**  Convert the list of LangChain messages (e.g., HumanMessage, AIMessage, and ToolMessage) into the format expected by Ragas, so the evaluation framework can understand and process them properly.
+
+To convert the list of LangChain messages into a format suitable for Ragas evaluation, you have two options:
+#### Option 1: Use the Implementation Provided by Ragas
+Ragas has already provided a function `convert_to_ragas_messages` that can be used to convert LangChain messages into the format expected by Ragas. Here's how you can use it:
+
+
+```python
+from ragas.integrations.langgraph import convert_to_ragas_messages
+
+# Assuming 'result["messages"]' contains the list of LangChain messages
+ragas_trace = convert_to_ragas_messages(result["messages"])
+```
+
+#### Option 2: Write a Custom Conversion Function
+Alternatively, you can implement your own conversion function by drawing inspiration from the code below. This function converts a list of LangChain messages (HumanMessage, AIMessage, ToolMessage) into Ragas-compatible message objects.
+
```

Comment:
why are we giving 2 options here? I think it's better to have them import it and make sure you add it in the Reference page and link it in this section. So users can check it out if they need to

Reply:
Understood and,

Could you please share the process for adding it to the reference page?

Reply:
refer to this file https://github.com/explodinggradients/ragas/blob/main/docs/references/integrations.md

and you can link to it like so [convert_to_ragas_message][ragas.integrations.langgraph.convert_to_ragas_message] 

does that help?

Reply:
Thank you for the clarification! Let me take a look at the file and review the details. I’ll get back to you shortly with any further questions or updates.

Reply:
I attempted to use the function [convert_to_ragas_message](ragas.integrations.langgraph.convert_to_ragas_message), but I encountered an error stating "Unable to open 'ragas.integrations.langgraph.convert_to_ragas_message'." I also tried using [convert_to_ragas_messages](ragas.integrations.langgraph), but the same issue occurred: "Unable to open 'ragas.integrations.langgraph'."

I have made some changes please have a look at them